src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -61,64 +67,102 @@ private LogParserPublisher(final boolean unstableOnWarning,
+            boolean preformattedHtml = !((DescriptorImpl) getDescriptor()).getLegacyFormatting();
             final FilePath parsingRulesFile;
             if (useProjectRule) {
-                parsingRulesFile = new FilePath(build.getWorkspace(),
-                        projectRulePath);
+                parsingRulesFile = new FilePath(workspace, projectRulePath);
             } else {
                 parsingRulesFile = new FilePath(new File(parsingRulesPath));
             }
-            final LogParserParser parser = new LogParserParser(
-                    parsingRulesFile, preformattedHtml, launcher.getChannel());
+            final LogParserParser parser = new LogParserParser(parsingRulesFile, preformattedHtml, launcher.getChannel());
             // Parse the build's log according to these rules and get the result
-            result = parser.parseLog(build);
+            result = parser.parseLog(run);
 
             // Mark build as failed/unstable if necessary
             if (this.failBuildOnError && result.getTotalErrors() > 0) {
-                build.setResult(Result.FAILURE);
+                run.setResult(Result.FAILURE);
             } else if (this.unstableOnWarning && result.getTotalWarnings() > 0) {
-                build.setResult(Result.UNSTABLE);
+                run.setResult(Result.UNSTABLE);
             }
         } catch (IOException e) {
             // Failure to parse should not fail the build - but should be
             // handled as a serious error.
             // This should catch all process problems during parsing, including
             // parser file not found..
-            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + build, e);
+            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + run, e);
             result.setFailedToParseError(e.toString());
         } catch (NullPointerException e) {
             // in case the rules path is null
-            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + build, e);
+            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + run, e);
             result.setFailedToParseError(e.toString());
-            build.setResult(Result.ABORTED);
+            run.setResult(Result.ABORTED);
         } catch (InterruptedException e) {
-            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + build, e);
+            logger.log(Level.SEVERE, LogParserConsts.CANNOT_PARSE + run, e);
             result.setFailedToParseError(e.toString());
-            build.setResult(Result.ABORTED);
+            run.setResult(Result.ABORTED);
         }
 
         // Add an action created with the above results
-        final LogParserAction action = new LogParserAction(build, result);
-        build.getActions().add(0, action);
-
-        return true;
+        final LogParserAction action = new LogParserAction(run, result);
+        run.addAction(action);

```

Comment:
The only reason to add an `Action` to a `Run` at index 0 that I can think of would be if you were expecting multiple actions of the same class to be added (without coalescing data!), and cared about order. But that could not have been the case here since the action was added by a `Publisher`, which on an `AbstractBuild` can only run once.